### PR TITLE
types: make UseEmptyDir a pointer

### PR DESCRIPTION
**Pull request by kragniz**
Co-authored-by: Richard Wall <richard.wall@jetstack.io>

### CODE REVIEW

**Comment by wallrj** [comment]
We plan to move all API helper functions into a separate, versioned, module; rather than add them as methods of the  API.
This is  consistent with how Kubernetes organise their own helpers: https://github.com/kubernetes/kubernetes/tree/master/pkg/apis/core/v1/helper

**Comment by sebbonnet** [comment]
Part of #33

File: cassandra-operator/pkg/apis/cassandra/v1alpha1/types.go [hunk 1]
```diff
@@ -36,8 +36,8 @@ type CassandraSpec struct {
 	Datacenter *string `json:"datacenter,omitempty"`
 	Racks      []Rack  `json:"racks"`
 	// +optional
-	UseEmptyDir bool `json:"useEmptyDir"`
-	Pod         Pod  `json:"pod"`
+	UseEmptyDir *bool `json:"useEmptyDir"`
```
[the remark this answers is not in the document]
Comment (kragniz) [reply cassandra-operator/pkg/apis/cassandra/v1alpha1/types.go]: Oops, fixed

File: cassandra-operator/pkg/operator/operations/adjuster/adjuster.go [hunk 2]
```diff
@@ -191,7 +191,11 @@ func (r *Adjuster) ensureChangeIsAllowed(oldCluster, newCluster *v1alpha1.Cassan
 		return fmt.Errorf("changing image is forbidden. The image used will continue to be '%v'", currentImage)
 	}
 	if !reflect.DeepEqual(oldCluster.UseEmptyDir, newCluster.UseEmptyDir) {
-		return fmt.Errorf("changing useEmptyDir is forbidden. The useEmptyDir used will continue to be '%v'", oldCluster.UseEmptyDir)
+		useEmptyDir := false
```
Comment (sebbonnet) [inline cassandra-operator/pkg/operator/operations/adjuster/adjuster.go]: why not using the `v1alpha1helpers.UseEmptyDir` helper function?

**Comment by kragniz** [comment]
Contained in #93.

**Pull request closed without merging**